Backend/index.js

```diff
@@ -326,15 +332,23 @@ app.post('/create-appointment', async (req, res) => {
         const slots = schedule.slots;
         const status = schedule.status;
         const slotIndex = slots.findIndex(slot => new Date(slot).toTimeString().slice(0, 5) === time);
+
+        const requiredSlots = Math.ceil(service.duration * 60 / 30);
```

Comment:
service.duration unit is hours.
For 30 mins, it will be 0.5 * 60/ 30 = 1, in this case required slots is 1
For 2 hours, it will be 2 * 60/30 = 4, in this case required slots is 4